oteapi/strategies/parse/image.py

```diff
@@ -93,53 +147,85 @@ def get(
+
+        config = self.parse_config.configuration
+        crop = config.crop if config.crop else session.get("imagecrop")
 
         mime_format = self.parse_config.mediaType.split("/")[1]
-        image_format = self._map_mime_to_format()
+        image_format = SupportedFormats[mime_format].value
+
+        # Proper download configurations
+        conf = self.parse_config.dict()
+        conf["configuration"] = config.download_config
+        download_config = ResourceConfig(**conf)
 
-        # Retrieve image file
-        download_config = ResourceConfig(**self.parse_config.dict())
-        del download_config.configuration
         downloader = create_strategy("download", download_config)
         session.update(downloader.initialize(session))
-        cache_key = downloader.get(session).get("key", "")
 
-        cache = DataCache(self.parse_config.configuration.datacache_config)
+        downloader = create_strategy("download", download_config)
+        output = downloader.get(session)
+        session.update(output)
+
+        if config.datacache_config and config.datacache_config.accessKey:
+            cache_key = config.datacache_config.accessKey
+        elif "key" in output:
+            cache_key = output["key"]
+        else:
+            RuntimeError("no datacache key provided to downloaded content")
+
+        cache = DataCache(config.datacache_config)
 
         # Treat image according to filter values
         with cache.getfile(cache_key, suffix=mime_format) as filename:
-            image = Image.open(filename, formats=[image_format]).crop(
-                self.parse_config.configuration.crop
+            image = Image.open(filename, formats=[image_format])
+            if crop:
+                image = image.crop(crop)
+            if config.image_mode:
+                image = image.convert(mode=config.image_mode)
+
+            if image_format == "GIF":
+                if image.info.get("version", b"").startswith(b"GIF"):
+                    image.info.update(
+                        {"version": image.info.get("version", b"")[len(b"GIF") :]}
+                    )
+
+            # Use the buffer protocol to store the image in the datacache
+            data = np.asarray(image)
+            image_key = cache.add(
+                data,
+                key=config.image_key,
+                tag=str(id(session)),
             )
 
-        if image_format == "GIF":
-            if image.info.get("version", b"").startswith(b"GIF"):
-                image.info.update(
-                    {"version": image.info.get("version", b"")[len(b"GIF") :]}
+            if image.mode == "P":
+                image_palette_key = cache.add(
+                    np.asarray(image.getpalette()), tag=str(id(session))
                 )
+            else:
+                image_palette_key = None
+
+            # The session must be json serialisable - filter out all
+            # non-json serialisable fields in image.info
```

Comment:
I think we should consider providing a custom JSON encoder in `oteapi-core` that can be used in `oteapi-services` and similar.

Reply:
Yes, that might be useful. We could do it in the same way as for the data cache...

Reply:
Added issue 79 to oteapi-services about this.